### PR TITLE
Replace Cilium LoadBalancer with MetalLB

### PR DESCRIPTION
**Pull request by bschimke95**
This PR replaces the Cilium LoadBalancer with MetalLB in the default Canonical Kubernetes configuration. The Cilium LoadBalancer had a bunch of shortcomings, e.g. IPv6 support.

### CODE REVIEW

**Review by berkayoz — APPROVED** [review]
LGTM, we can merge after verifying we can reach ingress or gateway services through a loadbalancer ip assigned by metallb.

**Review by louiseschmidtgen — COMMENTED** [review]
Nice work Ben! little suggestion

File: tests/integration/tests/test_util/util.py [hunk 1]
```diff
@@ -428,3 +429,27 @@ def _maj_min(version: str):
...
+    net = ipaddress.IPv4Network(parent_cidr, False)
+
+    # Starting from the first IP address from the parent cidr,
+    # we search for a /30 cidr block(4 total ips, 2 available)
+    # that doesn't contain the excluded ips to avoid collisions
+    # /30 because this is the smallest CIDR cilium hands out IPs from
+    for i in range(4, 255, 4):
+        lb_net = ipaddress.IPv4Network(f"{str(net[0]+i)}/30", False)
+
+        contains_excluded = False
```
[the remark this answers is not in the document]
Comment (bschimke95) [reply tests/integration/tests/test_util/util.py:445]: the `excluded_ips` are single IPs. A CIDR contains a range of IPs.
Your suggestion implies that the excluded parameter contains CIDRs, right?

File: .github/workflows/integration.yaml [hunk 2]
```diff
@@ -79,6 +79,10 @@ jobs:
     steps:
       - name: Check out code
         uses: actions/checkout@v4
+      - name: Setup tmate session
+        uses: mxschmitt/action-tmate@v3
+        with:
+          detached: true
```
Comment (eaudetcobello) [inline .github/workflows/integration.yaml:85]: Revert before merge

File: tests/integration/tests/test_util/util.py [hunk 3]
```diff
@@ -212,12 +213,12 @@ def wait_until_k8s_ready(
 
 def wait_for_dns(instance: harness.Instance):
     LOG.info("Waiting for DNS to be ready")
-    instance.exec(["k8s", "x-wait-for", "dns"])
+    instance.exec(["k8s", "x-wait-for", "dns", "--timeout", "20m"])
```
Comment (eaudetcobello) [inline tests/integration/tests/test_util/util.py:216]: Can we make this less than 20m? I don't think it looks too good on us to have long waits like this. (Just to be clear, this is not your fault).

File: .github/workflows/integration.yaml [hunk 4]
```diff
@@ -103,12 +103,14 @@ jobs:
           TEST_LXD_IMAGE: ${{ matrix.os }}
           TEST_INSPECTION_REPORTS_DIR: ${{ github.workspace }}/inspection-reports
         run: |
-          cd tests/integration && sg lxd -c 'tox -e integration'
+          cd tests/integration && sg lxd -c 'tox -e integration -- -k test_loadbalancer'
```
Comment (HomayoonAlimohammadi) [inline .github/workflows/integration.yaml:106]: Just a reminder to delete before merging.

**Pull request merged**